lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -591,6 +591,7 @@ public function setResultCacheProfile(?QueryCacheProfile $profile = null)
      */
     public function setResultCacheDriver($resultCacheDriver = null)
     {
+        /** @phpstan-ignore-next-line */
```

Comment:
What error are we ignoring here, and why do it that way? If future versions of PHPStan find more things here, we are going to ignore them too because of this, right?

Reply:
We're ignoring the if because it's doing a FC check for the argument type, which is for the same type that is declared in the PHPDoc, so PHPStan is complaining that the `if` clause will be always true.

Maybe we can specify the kind of error in the annotation? Psalm has that functionality, I don't know if PHPStan does.

Reply:
Well we could always add that error to the baseline, couldn't we?

Reply:
I think I understand why you're doing it this way: there is no plan to remove that ignore rule ever. Sounds fair 👍